### PR TITLE
Feature/webops 320 annotations

### PR DESCRIPTION
**Pull request by pingers**
Provides a key value field to collect annotations for routes and pass them to the OpenShift client. 

See also https://github.com/universityofadelaide/openshift-client/pull/23

### CODE REVIEW

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php [hunk 1]
```diff
@@ -314,4 +314,25 @@ public function getSite(NodeInterface $environment) {
...
+   *
+   * @param \Drupal\node\NodeInterface $environment
+   *   Environment entity.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|bool
+   *   The environment type term or FALSE.
+   */
+  public function getEnvironmentType(NodeInterface $environment) {
+    if (isset($environment->field_shp_environment_type->target_id)) {
+      return $environment->get('field_shp_environment_type')
```
[the remark this answers is not in the document]
Comment (pingers) [reply web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: Entity... magic pls.

**Comment by pingers** [comment]
TODO
- inject services liberally
- config export to support updating field on existing shepherd deployments

**Comment by bart88** [comment]
Argh 494 files changed. Reading the story for this, this PR should only have changes related to annotations.

**Comment by bart88** [comment]
@pingers Merged #107 into this branch so the tests can run. 😄

**Comment by pingers** [comment]
Well, yeah... but it's the first thing since we released that needed update hooks... I'm not writing update hooks if we're moving to a different config management method. Just look at the prior commits for annotations related stuff... I know it's annoying.

**Pull request merged**